include/mbgl/style/transition_options.hpp

```diff
@@ -6,9 +6,21 @@
 namespace mbgl {
 namespace style {
 
-struct TransitionOptions {
+class TransitionOptions {
+public:
     optional<Duration> duration = {};
     optional<Duration> delay = {};
+
+    TransitionOptions defaults(const TransitionOptions& defaults) const {
+        return {
+            duration ? duration : defaults.duration,
+            delay ? delay : defaults.delay
+        };
+    }
```

Comment:
I was modeling on [`_.defaults`](https://lodash.com/docs/4.17.1#defaults) but would be fine with renaming it to `reverseMerge` or something like that.